pluggy/__init__.py

```diff
@@ -664,7 +664,11 @@ def _add_hookimpl(self, hookimpl):
     def __repr__(self):
         return "<_HookCaller %r>" % (self.name,)
 
-    def __call__(self, **kwargs):
+    def __call__(self, *args, **kwargs):
+        if args:
+            raise TypeError(
+                "hookimpl takes keyword arguments only".format(self.name)
```

Comment:
that message is incorrect and cant hope to be formatted that way

i propose `"hook calling supports only keyword arguments"` as sole message